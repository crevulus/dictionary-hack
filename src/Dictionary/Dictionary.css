.dictionary-div .data {
  width: 70%;
  margin: auto;
  margin-top: 10px;
  border: 1px solid black;
  border-radius: 5px;
  background-color: white;
  box-shadow: 2px 2px 2px #555;
}

.entry {
  margin-bottom: -33px;
}

.search-text {
  color: white;
}

.word {
  font-size: 3em;
  font-weight: 200;
  margin-bottom: -8px;
}

.pronunciation {
  font-weight: 800;
}

.category {
  font-style: italic;
}