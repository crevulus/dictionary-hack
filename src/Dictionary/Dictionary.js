import React, { Component } from 'react';
import './Dictionary.css';

class Dictionary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      entry: 'Search for something!',
      word: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.fetchData = this.fetchData.bind(this);
  }

  // Registers search term on each key press
  handleChange(event) {
    this.setState({ searchValue: event.target.value });
  }

  // Resets search box
  handleSubmit(event) {
    event.preventDefault();
    this.setState({
      searchValue: '',
    });
  }

  // Calls API directly
  fetchData() {
    // store lexical content at time of creation
    const that = this;
    const { searchValue } = that.state;
    fetch(`https://wordsapiv1.p.rapidapi.com/words/${searchValue}`, {
      method: 'GET',
      withCredentials: true,
      headers: {
        'x-rapidapi-key': 'a2c92537d6mshed3836357d1a824p120053jsn7c815bcc2e2d',
      },
    })
    // Append to virtual DOM
      .then((resp) => resp.json())
      .then((data) => {
        that.setState({ entry: '' });
        that.setState({ word: data.word });
        that.setState({ pronunciation: data.pronunciation.all });
        that.setState({ category: data.results[0].partOfSpeech });
        that.setState({ definition: data.results[0].definition });
        console.log(data);
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    const {
      entry,
      word,
      pronunciation,
      category,
      definition,
      searchValue,
    } = this.state;
    return (
      <div className="dictionary-div">
        <form onSubmit={this.handleSubmit}>
          <label className="search-text">
            Search for a word:
            <br />
            <input type="text" label="Search" value={searchValue} onChange={this.handleChange} />
          </label>
          <input className="submit" type="submit" value="Search" onClick={this.fetchData} />
        </form>
        <div className="data">
          <p className="entry">{entry}</p>
          <p className="word">{word}</p>
          <p className="pronunciation">{pronunciation}</p>
          <p className="category">{category}</p>
          <p className="definition">{definition}</p>
        </div>
      </div>
    );
  }
}

export default Dictionary;

// fetchData = (searchValue) => {
//   const that = this;
//   fetch(`http://localhost:3001/`, {
//     method: 'GET',
//     headers: {
//       Accept: 'application/json',
//     },
//   })
//   .then((resp) => resp.json())
//   .then((data) => console.log(data));
// }
