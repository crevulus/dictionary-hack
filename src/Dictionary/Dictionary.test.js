/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import Dictionary from './Dictionary';
import renderer from 'react-test-renderer';

test('renders component', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Dictionary></Dictionary>, div);
});

test('matches snapshot', () => {
  const snap = renderer.create(<Dictionary></Dictionary>).toJSON();
  expect(snap).toMatchSnapshot();
});