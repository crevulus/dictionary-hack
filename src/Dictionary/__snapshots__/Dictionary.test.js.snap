// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="dictionary-div"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="search-text"
    >
      Search for a word:
      <br />
      <input
        label="Search"
        onChange={[Function]}
        type="text"
      />
    </label>
    <input
      className="submit"
      onClick={[Function]}
      type="submit"
      value="Search"
    />
  </form>
  <div
    className="data"
  >
    <p
      className="entry"
    >
      Search for something!
    </p>
    <p
      className="word"
    >
      
    </p>
    <p
      className="pronunciation"
    />
    <p
      className="category"
    />
    <p
      className="definition"
    />
    <p
      className="synonyms"
    />
  </div>
</div>
`;
